fling/dataset/domainnet.py

```diff
@@ -0,0 +1,93 @@
+import torch.utils.data as data
+import torchvision.transforms as transforms
+import os
+from PIL import Image
+import numpy as np
+
+from fling.utils.registry_utils import DATASET_REGISTRY
+
+
+@DATASET_REGISTRY.register('domainnet')
+class DomainNetDataset(data.Dataset):
+    r"""
+    Implementation for DomainNet dataset. 
+    """
+    def __init__(self, cfg: dict, domain: str,  train: bool): # cfg can be loaded
+        super(DomainNetDataset, self).__init__()
+        self.train_transform = transforms.Compose([
+        transforms.Resize([256, 256]),
+        transforms.RandomHorizontalFlip(),
+        transforms.RandomRotation((-30, 30)),
+        transforms.ToTensor()
+    ])
+        self.test_transform = transforms.Compose([
+        transforms.Resize([256, 256]),
+        transforms.ToTensor()
+    ])
+        self.train = train
+        self.cfg = cfg
+        self.train_imgs, self.train_labels, self.test_imgs, self.test_labels = \
+        load_domainnet(base_dir=self.cfg.data.base_dir, domain=domain, train_num=500, test_num=300000)
+
+    def __len__(self):
+        if self.train:
+            return len(self.train_labels)
+        else: 
+            return len(self.test_labels)
+        
+    def __getitem__(self, index: int) -> dict:
+        if self.train:
+            img, label = self.train_imgs[index], self.train_labels[index]
+            if len(img.split()) != 3:
+                img = transforms.Grayscale(num_output_channels=3)(img)
```

Comment:
Add this data transform to ``self.transform``, depending on the name of this domain.